llrpConnector/src/main/java/io/vantiq/extsrc/llrpConnector/LLRPConnector.java

```diff
@@ -0,0 +1,1546 @@
+     * Running as a thread to be able to pause and wait for responses to messages
+     * which is received on blocking reads.
+     *
+     */
+    private void connectAndSetupReader () {
+
+        Thread setup = new Thread(() -> {
+            try {
+
+                systemPrint("connectAndSetupReader started : " + Thread.currentThread().getName());
+                sendLogMessage(INFO, "Attempting to connect to the the Reader");
+
+                // If connected, nothing to do
+                if (readerOffline && !vantiqSourceOffline) {
+
+                    try {
+                        // Clear any messages in the queue, keep tag messages
+                        queue.clear();
+
+                        // Try to establish a connection to the reader
+                        connection = new Socket(hostname, readerPort);
+                        out = new DataOutputStream(connection.getOutputStream());
+
+                        // Start up the ReaderThread to read messages form socket to Console
+                        rt = new ReadThread(connection);
+                        rt.start();
+
+                        // ReaderEventNotificationData is sent by the Reader for a connection attempt,
+                        // though can be sent for other reader events. List of events:
+                        //		<HoppingEventParameter>,
+                        //		<GPIEvent Parameter>,
+                        //		<ROSpecEvent Parameter>,
+                        //		<ReportBufferLevelWarningEvent Parameter>,
+                        //		<ReportBufferOverflowErrorEvent Parameter>,
+                        //		<ReaderExceptionEvent Parameter>,
+                        //		<RFSurveyEvent Parameter>,
+                        //		<AISpecEvent Parameter>,
+                        //		<AntennaEvent Parameter>,
+                        //		<ConnectionAttemptEvent Parameter>,
+                        //		<ConnectionCloseEvent Parameter> - unsolicited close by the reader
+                        // Since we just tried to connect, we are expecting a ConnectionAttempEvent.
+                        //	Possible Values:
+                        //		  Value Definition
+                        //		  ----- ----------
+                        //			0	Success
+                        //			1	Failed (a Reader initiated connection already exists)
+                        //			2	Failed (a Client initiated connection already exists)
+                        //			3	Failed (any reason other than a connection already exists
+                        //			4	Another connection attempted
+                        //
+                        // We will skip all responses from the reader until a ConnectionAttempEvent is
+                        // received.
+                        LLRPMessage m = getNextMessage("CONNECTION_ATTEMPT");
+                        READER_EVENT_NOTIFICATION readerEventNotification = (READER_EVENT_NOTIFICATION) m;
```

Comment:
Might be good here to check if that cast is possible.  

```
if (m instanceof READER_EVENT_NOTIFICATON) { ...
```
otherwise we risk crash from class cast exception, don't we?

Reply:
Well, getNextMessage will only respond with a READER_EVENT_NOTIFICATION (or whatever message it is requesting).